patches/api/0466-Add-BeehiveNectarDepositEvent.patch

```diff
@@ -0,0 +1,127 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: maxcom1 <46265094+maxcom1@users.noreply.github.com>
+Date: Fri, 8 Mar 2024 18:05:05 +0100
+Subject: [PATCH] Add BeehiveNectarDepositEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/block/BeehiveNectarDepositEvent.java b/src/main/java/io/papermc/paper/event/block/BeehiveNectarDepositEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..8feaee4bb79fbf4c03e11f383d4b79f2f67883af
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/block/BeehiveNectarDepositEvent.java
+@@ -0,0 +1,115 @@
++package io.papermc.paper.event.block;
++
++import org.bukkit.block.Block;
++import org.bukkit.entity.Bee;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.block.BlockEvent;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Fired when Bee's nectar is deposited into the Beehive.
++ * <p>
++ * The event is fired right after the bee is released, only if there is a space in the hive for new nectar
++ */
++public class BeehiveNectarDepositEvent extends BlockEvent implements Cancellable {
```

Comment:
Generally, this entire event is not needed.
EntityChangeBlockEvent already exposes
 a) the changing block in its old state and hence a way to get the old honey level
 b) the block data of the block after the change, which can be used to get and set the new honey level
 c) point B also allows to compute the deposited nectar
 d) the bee is available as the entity of the entity change block event.
 
The API addition here is hence not really needed and we can instead just call EntityChangeBlockEvent.

The API patch can hence be completely dropped and the server patch changed to call EntityChangeBlockEvent.
If you are comfortable enough, you can also merge the server patch changes into the
`Fire EntityChangeBlockEvent in more places` patch.
Make sure to add yourself as a co-author there :+1: